core/src/main/java/org/mule/runtime/core/DefaultMuleMessageBuilder.java

```diff
@@ -44,13 +52,13 @@
     private String id;
     private String rootId;
     private String correlationId;
-    private int correlationSequence;
-    private int correlationGroupSize;
+    private int correlationSequence = -1;
+    private int correlationGroupSize = -1;
     private Object replyTo;
     private ExceptionPayload exceptionPayload;
 
-    private Map<String, TypedValue<Serializable>> inboundProperties = new HashMap<>();
-    private Map<String, TypedValue<Serializable>> outboundProperties = new HashMap<>();
+    private Map<String, TypedValue<Serializable>> inboundProperties = new CaseInsensitiveMapWrapper<>(HashMap.class);
```

Comment:
We use CaseInsensitiveHashMap in DefaultMuleMessage.  We either need to continue to us this, or deprecate/remove it if the use of CaseInsensitiveMapWrapper is preferred.